homeassistant/components/proxmoxve/binary_sensor.py

```diff
@@ -1,57 +1,61 @@
 """Binary sensor to read Proxmox VE data."""
 
+import logging
+
 from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.typing import HomeAssistantType
 from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
 
-from . import COORDINATORS, DOMAIN, PROXMOX_CLIENTS, ProxmoxEntity
+from . import COORDINATORS, DOMAIN, ProxmoxEntity
+
+_LOGGER = logging.getLogger(__name__)
 
 
-async def async_setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_entry(
+    hass: HomeAssistantType, config_entry: ConfigEntry, async_add_entities
+) -> None:
     """Set up binary sensors."""
-    if discovery_info is None:
-        return
 
     sensors = []
 
-    for host_config in discovery_info["config"][DOMAIN]:
-        host_name = host_config["host"]
-        host_name_coordinators = hass.data[DOMAIN][COORDINATORS][host_name]
+    _LOGGER.info("Setting up ProxmoxVE platform")
 
-        if hass.data[PROXMOX_CLIENTS][host_name] is None:
-            continue
+    host_name = config_entry.data["host"]
+    host_name_coordinators = hass.data[DOMAIN][COORDINATORS][host_name]
```

Comment:
It would be better to fetch this on entry ID

```suggestion
    coordinators = hass.data[DOMAIN][entry.entry_id][COORDINATORS]
```